### ISSUE TITLE
Firefox dotted inner focus should be removed from all elements where we define a custom focus state

### ISSUE THREAD

**Issue by lovemecomputer** [carbon-design-system/carbon#1526]
Firefox requires an additional rule `::-moz-focus-inner` to remove the additional dotted-line focus state. This should be applied to all input/etc elements we can fine where we are already defining a focus state—the doubled outlines looks confusing.

Example solution from https://stackoverflow.com/questions/71074/how-to-remove-firefoxs-dotted-outline-on-buttons-as-well-as-links :

```css
button::-moz-focus-inner {
  border: 0;
}
```

**Buttons:**
![screen shot 2018-12-04 at 10 54 06 am](https://user-images.githubusercontent.com/3846271/49458984-996c0500-f7b3-11e8-99f1-641c5acdcb8b.png)

**Select:**
<img width="306" alt="screen shot 2018-12-04 at 10 58 03 am" src="https://user-images.githubusercontent.com/3846271/49459096-ddf7a080-f7b3-11e8-8fda-053fe760a14b.png">

 **Code block:**
<img width="865" alt="screen shot 2018-12-04 at 10 55 51 am" src="https://user-images.githubusercontent.com/3846271/49459110-e64fdb80-f7b3-11e8-9157-4acfb241cf0b.png">

---

Presumably this will be an issue we need to solve for all input elements, but this last code block example shows we may need to look a little more thoroughly

**Comment by asudoh** [comment]
Nice finding @lovemecomputer! One option we may have is using our reset style, if we find doing so is safe enough.

**Comment by emyarod** [comment]
would this apply to `<select>`s as well? dealing with Firefox outlines in `<select>` are a bit trickier

**Comment by lovemecomputer** [comment]
the official design ask would be to removed the additional dotted line anywhere we are doing are own stylings for focus, to avoid doubling-up. but ofc testing on a per-item and per-platform basis, we can just do the best we can :)
